### PR TITLE
refactor: [M3-6340] - MUI v5 Migration - `Components > FileUploader`

### PR DESCRIPTION
**Pull request by hana-linode**
## Description 📝
- Fix bug where the upload arrow wasn't animating
- Migrate styles for the FileUploader component and code pattern updates

## Preview 📷
Before

https://github.com/linode/manager/assets/115299789/b9101542-592b-4d2f-ba31-f0c789c472df

After

https://github.com/linode/manager/assets/115299789/9857eced-d717-4e1c-a590-4ddd8d3572ad

## How to test 🧪
Check the success/error states of the File Upload in `/images/create/upload` and in an Object Storage bucket, ensure there are no regressions

### CODE REVIEW

File: packages/manager/src/components/FileUploader/FileUploader.tsx [hunk 1]
```diff
@@ -407,78 +322,54 @@ const FileUploader: React.FC<CombinedProps> = (props) => {
...
     ? 'To upload an image, complete the required fields.'
     : 'You can browse your device to upload an image file or drop it here.';
 
   return (
-    <div
-      className={classNames({
-        [classes.root]: UploadZoneActive,
+    <StyledDropZoneDiv
+      {...getRootProps({
+        className: `${dropZoneClasses}`,
```
Comment (jaalah-akamai) [inline packages/manager/src/components/FileUploader/FileUploader.tsx:334]: You can just make those classes props on this styled component:
```
<StyledDropZoneDiv
    isDragActive={isDragActive}
    isDragAccept={isDragAccept}
    isDragReject={isDragReject}
    dropzoneDisabled={dropzoneDisabled}
    uploadInProgressOrFinished={uploadInProgressOrFinished}
  />
```

```
const StyledDropZoneDiv = styled(Box)(({ isDragActive, isDragAccept, isDragReject, dropzoneDisabled, uploadInProgressOrFinished }) => ({
  borderColor: isDragActive || isDragAccept ? theme.palette.primary.light : (isDragReject ? theme.color.red : (dropzoneDisabled || uploadInProgressOrFinished ? '#888' : undefined)),
  backgroundColor: isDragActive ? theme.color.white : theme.palette.primary.main,
}));
```
Check if you need `getRootProps` anymore...

File: packages/manager/src/components/FileUploader/FileUploader.styles.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import Button from 'src/components/Button';
+import { styled } from '@mui/material/styles';
+import { isPropValid } from 'src/utilities/isPropValid';
+import Typography from 'src/components/core/Typography';
+import { makeStyles } from 'tss-react/mui';
+import { Theme } from '@mui/material/styles';
+
+export const StyledDropZoneDiv = styled('div')(({ theme }) => ({
```
Comment (abailly-akamai) [inline packages/manager/src/components/FileUploader/FileUploader.styles.ts:8]:
```suggestion
export const StyledDropZoneDiv = styled('div', { label: 'StyledDropZoneDiv' })(({ theme }) => ({
```

Let's please add labels for all styled components. Makes it easier to debug in the browser!

File: packages/manager/src/components/FileUploader/FileUploader.styles.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  color: theme.palette.primary.main,
+  height: '100%',
+  maxHeight: 400,
+  marginTop: theme.spacing(2),
+  minHeight: 140,
+  outline: 'none',
+  overflow: 'auto',
+  padding: theme.spacing(),
+  transition: theme.transitions.create(['border-color', 'background-color']),
+}));
```
Comment (abailly-akamai) [inline packages/manager/src/components/FileUploader/FileUploader.styles.ts:26]: alphabetically ordering long lists of CSS properties would help readability here and on other Styled components

File: packages/manager/src/features/ObjectStorage/ObjectUploader/FileUpload.tsx [hunk 4]
```diff
@@ -104,8 +104,8 @@ export const FileUpload = React.memo((props: FileUploadProps) => {
               width={22}
             />
           ) : (
-            <UploadPending
-              className={`${classes.iconRight} ${classes.rotate}`}
+            <StyledUploadPending
+              className={classes.iconRight}
```
Comment (abailly-akamai) [inline packages/manager/src/features/ObjectStorage/ObjectUploader/FileUpload.tsx:108]: If we have a styled component here we shouldn't have to use `classes`

File: packages/manager/src/components/FileUploader/FileUploader.tsx (hunk 1 above)
Comment (hana-linode) [reply packages/manager/src/components/FileUploader/FileUploader.tsx:334]: @jaalah-akamai I tried doing something similar but the border colors don't seem to work w/o `getRootProps` which is why I left as is 🤔

Comment (jaalah-akamai) [reply packages/manager/src/components/FileUploader/FileUploader.tsx:334]: Just to clarify that the new interface would need to be applied to `StyledDropZoneDiv`:

**Review by jaalah-akamai — APPROVED** [review]
Looking good!

**Review by dwiley-akamai — APPROVED** [review]
LGTM 🚢

**Pull request merged**